### ISSUE TITLE
[Question] numpy version

### ISSUE THREAD

**Issue by imgarylai** [dmlc/gluon-nlp#636]
https://github.com/dmlc/gluon-nlp/blob/6ec0c843842dbb8d8caa1cadb73c009ad7ed965e/setup.py#L31-L33

Should we specify the numpy version in gluonnlp which aligns with mxnet? I put gluonnlp and mxent as dependencies in a python package together. In some case, this package installation fails because when this package is installed via `pip install`, the latest numpy seen in the gluonnlp is installed first. However, the latest numpy version is outer the version specified by mxnet. Since dependencies in a python package are not installed in the order specified in the setup.py, it can not be solved by the ordering the dependencies to install mxnet first.

I am trying to replicate the environment where I have this issue. However, I can't reproduce it successfully every time. Please let me know what you think.

**Comment by szha** [comment]
Good catch. I think a good way to resolve it is to adapt the mxnet compatibility to greater range. Let's raise an issue in mxnet about this too and link it there.

**Comment by szha** [comment]
apache/incubator-mxnet#14497
